pdudaemon/httpserver.py

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/python
+
+#  Copyright 2018 Matt Hart <matt@mattface.org>
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+#  MA 02110-1301, USA.
+
+
+try:
+    from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
+    import urlparse
+except ImportError:
+    from http.server import BaseHTTPRequestHandler, HTTPServer
+    import urllib.parse as urlparse
+import logging
+import time
+import sys
+import os
+from pdudaemon.dbhandler import DBHandler
+from pdudaemon.shared import drivername_from_hostname
+log = logging.getLogger(__name__)
```

Comment:
https://docs.python.org/2/library/logging.html#logger-objects "Multiple calls to getLogger() with the same name will always return a reference to the same Logger object."
So I think it's neater to do that in each class with `self.logger = ...` rather than have a global variable.